[v7] TS automatic imports in VSCode default to wrong path

## Bug Report

**Current Behavior**
When using operators in VSCode, the suggested import defaults to `rxjs/dist/types/operators`, failing in runtime.

This wasn't happening in v6, starts happening after upgrading to v7.

**Expected behavior**
VSCode imports from the correct `rxjs/operators` path

**Reproduction**

* create a simple playground
```sh
mkdir rxjs-playground
cd rxjs-playground
npm init -y
npm i rxjs@next
```

* open that folder in VSCode
* create a file foo.ts

```ts
interval(1000).pipe( // Correctly suggests `import { interval } from 'rxjs'`
   map(v => v*2) // Doesn't suggest any valid import, that's fine - it used to do that. Manually import it from `rxjs/operators`
)
```

* create a file bar.ts and type in:

```ts
interval(1000).pipe( // Correctly suggests `import { interval } from 'rxjs'`
   map(v => v*2) // Wrongly suggests `import { map } from 'rxjs/dist/types/operators'`
)
```

**Environment**
- RxJS version: 7.0.0-beta.12
- VSCode: 1.53.2
- Node: 12.18.3
- Typescript: 4.1.5

**Additional context/Screenshots**

![image](https://user-images.githubusercontent.com/5365487/109515914-eaa8f300-7aa7-11eb-8faf-73dd16ba54f4.png)

Note that after correcting it to `rxjs/operators` it correctly suggests adding the import from the right place

![image](https://user-images.githubusercontent.com/5365487/109516077-15934700-7aa8-11eb-9214-8738f6b82eb2.png)

Strangely, in the rc.1 version, I get "error TS2307: Cannot find module 'rxjs' or its corresponding type declarations.".
I am using TypeScript 4.2.4 + VSCode 1.55.2.

--- a/package.json
+++ b/package.json
@@ -5,7 +5,12 @@
   "main": "./dist/cjs/index.js",
   "module": "./dist/esm5/index.js",
   "es2015": "./dist/esm/index.js",
-  "types": "./dist/types/index.d.ts",
+  "types": "index.d.ts",
+  "typesVersions": {
+    ">=4.2": {
+      "*": ["dist/types/*"]
+    }
+  },
   "sideEffects": false,
   "exports": {
     ".": {

--- a/src/index.ts
+++ b/src/index.ts
@@ -1,3 +1,18 @@
+//////////////////////////////////////////////////////////
+// Here we need to reference our other deep imports
+// so VS code will figure out where they are
+// see conversation here:
+// https://github.com/microsoft/TypeScript/issues/43034
+//////////////////////////////////////////////////////////
+
+// tslint:disable: no-reference
+/// <reference path="./ajax/index.ts" />
+/// <reference path="./fetch/index.ts" />
+/// <reference path="./operators/index.ts" />
+/// <reference path="./testing/index.ts" />
+/// <reference path="./webSocket/index.ts" />
+// tslint:enable: no-reference
+
 /* Observable */
 export { Observable } from './internal/Observable';
 export { ConnectableObservable } from './internal/observable/ConnectableObservable';

